src/BaGet.Core/Services/IndexingService.cs

```diff
@@ -44,7 +46,9 @@ public class IndexingService : IIndexingService
                     var packageId = packageReader.NuspecReader.GetId();
                     var packageVersion = packageReader.NuspecReader.GetVersion();
 
-                    if (await _packages.ExistsAsync(packageId, packageVersion))
+                    var identity = new PackageIdentity(packageId, packageVersion);
+
+                    if (await _storage.ExistsAsync(identity))

```

Comment:
We should not fail uploads if the package already exists in storage. Consider the following scenario:

1. A customer starts a package upload
2. The package is uploaded to storage
3. BaGet throws an exception before the package's metadata is persisted to the database

If we fail uploads when the package already exists in storage, we are now in an unrecoverable state. An admin needs to go manually delete the package from storage. The solution to these problems is to make the database the single source of truth.